### ISSUE TITLE
Modifier keys.

### ISSUE THREAD

**Issue by MikuAuahDark** [love2d/love#1797]
Currently it's not possible to know the state of caps lock, num lock, and scroll lock. LOVE fires key callbacks when those are pressed but those keys also have "active" state.

The proposed function is `love.keyboard.getModifierState(table)`. The function returns `table` (or it creates new table if not specified like `love.graphics.getStats`) with these structures:

```
{
    capslock = true or false,
    numlock = true or false,
    scrolllock = true or false
}
```

The only left question is on which branch this feature should land. Should it goes to `main` or `12.0-development`.

**Comment by zorggn** [comment]
One more request to this is `AltGr` detection; SDL apparently supports that with the `mode` enum value, and it would be needed to differentiate the `AltGr` key pressed on keyboards (with layouts having it) compared to if someone just pressed/held `lctrl` **and** `ralt`, which is what gets detected with all other APIs löve provides.

**Comment by slime73** [comment]
I think it'd be more consistent / idiomatic if the function took an enum name and returned a boolean.

> Should it goes to main or 12.0-development.

12.0

**Comment by MikuAuahDark** [comment]
So, something like `love.keyboard.getModifierState(key)` or `love.keyboard.isActive(key)` where `key` accepts 3 of the modifier keys and return `false` for anything else?

I personally think the latter name is confusing though.

**Comment by slime73** [comment]
maybe `love.keyboard.isModifierDown` or  `love.keyboard.hasModifier`?

**Comment by MikuAuahDark** [comment]
Implemented in 321397d1dd86b2a2a308221609dd93308f2bd6cd.
